tests/manifests/test_configs_consistency.py

```diff
@@ -0,0 +1,706 @@
+                assert_exists_according_to_hook_weight(configmap, weight, validated_config.name)
+                # We do not consume ConfigMaps in render-config, their configuration
+                # will actually be consumed later by the container using the rendered-config
+                current_source_of_mount = MountedConfigMap.from_template(configmap, volume_mount)
+                if not is_matrix_tools_command(container_spec, "render-config"):
+                    validated_config.paths_consumers.append(ConfigMapPathConsumer.from_configmap(configmap))
+            elif "emptyDir" in current_volume:
+                # An empty dir can be mounted multiple times on a container if using subPath
+                # So we need to keep track of them, create them without any rendered output
+                # We fill up the rendered output available to this container on the next step
+                if current_volume["name"] in previously_mounted_empty_dirs:
+                    current_source_of_mount = previously_mounted_empty_dirs[current_volume["name"]]
+                    if "subPath" in volume_mount:
+                        current_source_of_mount.mount_point = "/".join(volume_mount["mountPath"].split("/")[:-1])
+                    else:
+                        current_source_of_mount.mount_point = volume_mount["mountPath"]
+                else:
+                    current_source_of_mount = MountedEmptyDir.from_template(
+                        current_volume["name"], volume_mount, deployable_details.content_volumes_mapping
+                    )
+                validated_config.mutable_empty_dirs[current_volume["name"]] = current_source_of_mount
+            elif "persistentVolumeClaim" in current_volume:
+                current_source_of_mount = MountedPersistentVolume.from_template(
+                        current_volume, volume_mount, deployable_details.content_volumes_mapping
+                    )
+            # If we have a subPath we filter the files using a SubPathMount
+            if "subPath" in volume_mount:
+                validated_config.sources_of_mounted_paths.append(SubPathMount(volume_mount["subPath"],
+                                                                            current_source_of_mount))
+            else:
+                validated_config.sources_of_mounted_paths.append(current_source_of_mount)
+
+        mounted_files = [
+            node_path(parent_mount, mount_node)
+            for source in validated_config.sources_of_mounted_paths
+            for parent_mount, mount_node in source.get_mounted_paths()
+        ]
+        assert len(mounted_files) == len(set(mounted_files)), (
+            f"Mounted files are not unique in {container_spec['name']}\n"
+            f"Duplicated files : { {item for item, count in Counter(mounted_files).items() if count > 1} }\n"
+            f"From Mounted Sources : {validated_config.sources_of_mounted_paths}"
+        )
```

Comment:
This feels like it should be its own test, not part of the construction of a `ValidatedContainerConfig`